composer/utils/file_helpers.py

```diff
@@ -60,6 +62,38 @@ def ensure_folder_is_empty(folder_name: Union[str, pathlib.Path]):
                 raise FileExistsError(f"{folder_name} is not empty; {os.path.join(root, file)} exists.")
 
 
+def ensure_folder_has_no_conflicting_files(folder_name: Union[str, pathlib.Path], filename: str, timestamp: Timestamp):
+    """Ensure that the given folder does not have any files conflicting with filename.
+
+    Args:
+        folder_name (str | pathlib.Path): The folder to inspect.
+        filename (str): The pattern string for potential files.
+        timestamp (Timestamp): Ignore any files that occur before the provided timestamp.
+
+    Raises:
+        FileExistsError: If ``folder_name`` contains any files matching the ``filename`` template before ``timestamp``.
+    """
+    # Prepare regex pattern by replacing f-string formatting with regex. Currently, we only capture and compare time values for
+    # epoch and batch.
```

Comment:
Is it possible to compare all timing values? Epoch, batch, sample, and token should be monotonically increasing. batch_in_epoch, sample_in_epoch, and token_in_epoch should be increasing if the epoch is the same.